src/api/source/geojson.js

```diff
@@ -0,0 +1,184 @@
+            for (let i = 0; i < numFeatures; i++) {
+                const feature = this._features[i];
+                if (feature.type === 'Feature') {
+                    const type = feature.geometry.type;
+                    const coordinates = feature.geometry.coordinates;
+                    if (!this._type) {
+                        this._type = type;
+                    } else if (this._type !== type) {
+                        throw Error(`Multiple types not supported: ${this._type}, ${type}`);
+                    }
+                    if (type === 'Point') {
+                        if (!geometry) {
+                            geometry = new Float32Array(numFeatures * 2);
+                        }
+                        const point = this._computePointGeometry(coordinates);
+                        geometry[2 * i + 0] = point.x;
+                        geometry[2 * i + 1] = point.y;
+                    }
+                    else if (type === 'LineString') {
+                        if (!geometry) {
+                            geometry = [];
+                        }
+                        const line = this._computeLineStringGeometry(coordinates);
+                        geometry.push([line]);
+                    }
+                    else if (type === 'MultiLineString') {
+                        if (!geometry) {
+                            geometry = [];
+                        }
+                        const multiline = this._computeMultiLineStringGeometry(coordinates);
+                        geometry.push(multiline);
+                    }
+                    else if (type === 'Polygon') {
+                        if (!geometry) {
+                            geometry = [];
+                        }
+                        const polygon = this._computePolygonGeometry(coordinates);
+                        geometry.push([polygon]);
+                    }
+                    else if (type === 'MultiPolygon') {
+                        if (!geometry) {
+                            geometry = [];
+                        }
+                        const multipolygon = this._computeMultiPolygonGeometry(coordinates);
+                        geometry.push(multipolygon);
+                    }
+                }
+            }
+            const dataframe = new Dataframe(
+                { x: 0, y: 0 },
+                1,
+                geometry,
+                properties,
+            );
+            dataframe.type = this._getDataframeType(this._type);
+            dataframe.active = true;
+            dataframe.size = numFeatures;
+            this._addDataframe(dataframe);
+        }
+    }
+
+    _getDataframeType(type) {
+        if (type === 'Point') return 'point';
+        if (type === 'LineString' || type === 'MultiLineString') return 'line';
+        if (type === 'Polygon' || type === 'MultiPolygon') return 'polygon';
+        return '';
+    }
+
+    _computePointGeometry(data) {
```

Comment:
rename data to `latlng` or `point`

Reply:
I prefer to keep data, because is not a point (`{x,y}`) it is an "encoded" array (`[0, 0]`)